service/handlers.go

```diff
@@ -981,6 +981,64 @@ func (h handler) updateTarget(w http.ResponseWriter, r *http.Request) {
 	fmt.Fprint(w, string(data))
 }
 
+// Lists tokens for a project
+func (h handler) listTokens(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	projectName := vars["projectName"]
+
+	l := h.requestLogger(r, "op", "list-tokens", "project", projectName)
+
+	level.Debug(l).Log("message", "validating authorization header for token list")
+	ah := r.Header.Get("Authorization")
+	a, err := credentials.NewAuthorization(ah)
+	if err != nil {
+		h.errorResponse(w, "error unauthorized, invalid authorization header format", http.StatusUnauthorized)
+		return
+	}
+	if err := a.Validate(a.ValidateAuthorizedAdmin(h.env.AdminSecret)); err != nil {
+		h.errorResponse(w, "error unauthorized, invalid authorization header", http.StatusUnauthorized)
+		return
+	}
+
+	ctx := r.Context()
+
+	_, err = h.dbClient.ReadProjectEntry(ctx, projectName)
+	if err != nil {
+		level.Error(l).Log("message", "error retrieving project", "error", err)
+		if errors.Is(err, upper.ErrNoMoreRows) {
+			h.errorResponse(w, "project does not exist", http.StatusNotFound)
+		} else {
+			h.errorResponse(w, "error retrieving project", http.StatusInternalServerError)
+		}
+		return
+	}
+
+	tokens, err := h.dbClient.ListTokenEntries(ctx, projectName)
+	if err != nil {
+		level.Error(l).Log("message", "error listing targets", "error", err)
+		h.errorResponse(w, "error listing targets", http.StatusInternalServerError)
+		return
+	}
+
+	resp := []responses.ListTokens{}
+	for _, tokenEntry := range tokens {
+		resp = append(resp, responses.ListTokens{
+			CreatedAt: tokenEntry.CreatedAt,
+			ProjectID: tokenEntry.ProjectID,
```

Comment:
Do we wanna just remove project id from the response since we know know it'll be the same for every token for that project? I know rn the response struct just mirrors the db struct but if we're not using the same actual type then we might as well only return the fields that are useful right?

Reply:
Yeah, that's a good idea. I'll remove the project field.